api/src/home/workflow/job_handlers/local_container/__init__.py

```diff
@@ -0,0 +1,98 @@
+import docker
+from docker.errors import DockerException
+from typing import Tuple, Optional
+from services.job_handler_interface import JobHandlerInterface, JobStatus
+from utils.logging import logger
+from uuid import uuid4
+from dmss_api.apis import DefaultApi
+import json
+
+import os
+
+_SUPPORTED_TYPE = "WorkflowDS/Blueprints/jobHandlers/Container"
+
+# class JobEntity(BaseModel):
+#     name: str
+#     image: str
+#     command: Optional[str]
+
+
+class Settings:
+    PUBLIC_DMSS_API: str = os.getenv("PUBLIC_DMSS_API", "http://dmss:5000")  # tood port 8000?
+
+
+dmss_api = DefaultApi()
+settings = Settings()
+dmss_api.api_client.configuration.host = settings.PUBLIC_DMSS_API
+
+
+class JobHandler(JobHandlerInterface):
+    """
+    A job handler to run local docker container. This is similar to the local_containers job handler in DMT, but uses
+    a different blueprint
+    """
+
+    def __init__(self, data_source: str, job_entity: dict, token: str, insert_reference: callable):
+        super().__init__(data_source, job_entity, token, insert_reference)
+        self.headers = {"Access-Key": token}
+        try:
+            self.client = docker.from_env()
+        except DockerException:
+            raise DockerException(
+                (
+                    "Failed to get a docker client. Docker must be installed on this host, or "
+                    + "the /var/run/docker.sock must be made available (volume mount)."
+                    + "Make sure you are aware of the serious security risk this entails."
+                )
+            )
+
+    def start(self) -> str:
+        try:
+            runnerEntity: dict = self.job_entity["runner"]
+            logger.info(
+                f"JobName: '{self.job_entity.get('_id', self.job_entity.get('name'))}'."
+                + " Starting Local Container job..."
+            )
+            logger.info(f"job_entity is {self.job_entity}")
+
+            logger.info(
+                "Creating container\n\t"
+                + f"Image: '{runnerEntity['image']}'\n\t"
+                + f"Command: {runnerEntity.get('command', 'None')}"
+            )
+            json_separators = (
+                ",",
+                ":",
+            )  # the json string cannot contain any whitespace when using as command for container
+            app_input = json.dumps(self.job_entity["applicationInput"], separators=json_separators)
+
+            if " " in app_input:
+                raise Exception("Cannot have any spaces in the applicationInput for local container job")
```

Comment:
there was a strange bug here - if the json string contains any spaces at all, the container fails... 
There probably is a better way to deal with this.